src/node.cc

```diff
@@ -1197,6 +1224,18 @@ enum encoding ParseEncoding(Isolate* isolate,
   }
 }
 
+
+enum encoding ParseEncoding(Isolate* isolate,
+                            Handle<Value> encoding_v,
+                            enum encoding _default) {
+  if (!encoding_v->IsString())
+    return _default;
+
+  node::Utf8Value encoding(isolate, encoding_v);
+
+  return ParseEncoding(*encoding, _default);
```

Comment:
Also, wouldn't it be cheaper to do the comparison in JS and just pass integers to C++ land?  I suspect that the biggest overhead is from the new[]/delete[] calls, not the string comparison.